exporter/awss3exporter/logbody_marshaler.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package awss3exporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awss3exporter"
+
+import (
+	"bytes"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type bodylogMarshaler struct{}
+
+func (*bodylogMarshaler) format() string {
+	return "log"
+}
+
+func newbodylogMarshaler() bodylogMarshaler {
+	return bodylogMarshaler{}
+}
+
+func (bodylogMarshaler) MarshalLogs(ld plog.Logs) ([]byte, error) {
+	buf := bytes.Buffer{}
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+
+		ills := rl.ScopeLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ils := ills.At(j)
+			logs := ils.LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				lr := logs.At(k)
+				body := lr.Body()
+				buf.WriteString(body.Str())
```

Comment:
this doesn't work well for values that are not of type string. Please write a few test cases with a boolean, int, or map body. Use `AsString()` instead of `Str()`.

Reply:
Thank you for sharing your insights. I have updated to `AsString()` and added the test cases to cover various data types, including boolean, number, and map. Please kindly review my latest commit.